lib/src/tree.rs

```diff
@@ -830,10 +830,6 @@ fn simplify_conflict(
         }
     }
 
-    // TODO: We should probably remove duplicate entries here too. So if we have
-    // {+A+A}, that would become just {+A}. Similarly {+B-A+B} would be just
```

Comment:
(Not a blocker) I'm a bit confused by what was happening here, perhaps because I only looked at this superficially. Apologies if my questions are nonsense.

To me, it seems like the thing we might or might not want to do at this point is to run another `trivial_merge` on new_adds and new_removes. That would resolve `{+A-B+A}` to `A`. If `B` is empty, that covers the `{+A + A}` case.

If this is correct, you could leave a comment here explaining that we could do it but choose not to.

Reply:
Good question! I think this is the same case as we talked about in https://github.com/martinvonz/jj/pull/1610#discussion_r1199823932. It's weird that we don't use `trivial_merge()` here. If we do, we will resolve "same change" conflicts. If we decide that that's the right thing to do at the hunk level (and ref level, etc.), then we should do it here too for consistency. For now, I've just dropped this change from the PR because it's quite unrelated to the rest. I'll try to remember to work on a separate PR for using `trivial_merge()` here.